### PR TITLE
README: document the security model

### LINKED ISSUE #181: Document `pip-audit`'s security model
The distinction between `pip-audit` and `pip-audit -r <INPUT>` might lead some people to believe that the latter is a security boundary, i.e. something you can safely run when you don't trust your packages.

This isn't true (and can't be true, for dependency resolution reasons), so we should document exactly what `pip-audit`'s security model and guarantees are. In particular, we should make sure people understand that running `pip-audit -r ...` is **roughly equivalent** to running `pip install -r ...`.

Ref: https://github.com/trailofbits/pip-audit/issues/180#issuecomment-985818011

### PR DESCRIPTION
**Pull request by woodruffw**
Closes #181.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,14 @@ of vulnerability reports.
 This project is developed by [Trail of Bits](https://www.trailofbits.com/) with
 support from Google. This is not an official Google product.
 
+- [Installation](#installation)
+- [Usage](#usage)
+- [Examples](#examples)
+- [Security Model](#security-model)
+- [Licensing](#licensing)
+- [Contributing](#contributing)
+- [Code of Conduct](#code-of-conduct)
```
Comment (di) [inline README.md:22]: This won't work on PyPI: https://github.com/pypa/readme_renderer/issues/169

File: README.md [hunk 2]
```diff
@@ -169,6 +177,43 @@ Found 2 known vulnerabilities in 1 packages
...
+state which ones it has skipped, as well as why it has skipped them.
+
+`pip-audit` is **not** a static code analyzer. It analyzes dependency trees,
+not code, and it it **cannot** guarantee that arbitrary dependency resolutions
+occur statically. To understand why this is, refer to Dustin Ingram's
+[excellent post on dependency resolution in Python](https://dustingram.com/articles/2018/03/05/why-pypi-doesnt-know-dependencies/).
+
+As such: you **must not** assume that `pip-audit` will **defend** you against
+malicious packages. In particular, it is **incorrect** to treat
+`pip-audit -r INPUT` as a "more secure" variant of `pip-audit`. For all extents
```
Comment (di) [inline README.md:212]:
```suggestion
`pip-audit -r INPUT` as a "more secure" variant of `pip-audit`. For all intents
```

File: README.md [hunk 3]
```diff
@@ -169,6 +177,43 @@ Found 2 known vulnerabilities in 1 packages
...
+and purposes, `pip-audit -r INPUT` is functionally equivalent to
+`pip install -r INPUT`, with a small amount of **non-security isolation** to
+conflicts with any of your local environments.
```
Comment (di) [inline README.md:215]:
```suggestion
avoid conflicts with any of your local environments.
```

File: README.md [hunk 4]
```diff
@@ -169,6 +177,43 @@ Found 2 known vulnerabilities in 1 packages
...
+## Security Model
+
+This section exists to describe the security assumptions you **can** and **must not**
+make when using `pip-audit`.
+
+TL;DR: **If you wouldn't `pip install` it, you should not `pip audit` it.**
+
+`pip-audit` is a tool for auditing Python environments for packages with
+*known vulnerabilities*. A "known vulnerability" is a flaw in a package that,
+if uncorrected, *might* allow a malicious actor to perform unintended actions.
```
Comment (di) [inline README.md:189]: ...and also has been publicly reported to the project maintainers.

File: README.md [hunk 5]
```diff
@@ -169,6 +177,43 @@ Found 2 known vulnerabilities in 1 packages
...
+
+For example: `somepackage==1.2.3` might have a *known vulnerability* in the form
+of a deserialization bug. An attacker *might* be able to exploit that
+vulnerability, depending on how your code (or your other dependencies) use
+`somepackage`.
+
```
Comment (di) [inline README.md:195]:
```suggestion
```

I think we can nuke this.

**Review by di — APPROVED** [review]
LGTM w/ requested changes.

File: README.md (hunk 5 above)
Comment (tetsuo-cpp) [reply README.md:195]: Yeah, I think it's already clear enough.

File: README.md (hunk 1 above)
Comment (tetsuo-cpp) [reply README.md]: 440029c

File: README.md (hunk 4 above)
Comment (tetsuo-cpp) [reply README.md]: 1b0e38f

**Pull request merged**